Generic objects, when realized to strings, raise TypeError on concatenation.

CrossHair's counterexample does not reproduce in this example:
```
def arithmetic_op(a, b):
    """
    post: True
    """
    result = a + b
    return result

# actual: /tmp/main.py:5: error: TypeError:  when calling arithmetic_op('', '')
# expect: counterexamples like arithmetic_op('', 0)
```

See also the original report in #234

--- a/crosshair/libimpl/builtinslib.py
+++ b/crosshair/libimpl/builtinslib.py
@@ -3160,6 +3160,7 @@ class LazyIntSymbolicStr(AnySymbolicStr, CrossHairValue):
     @classmethod
     def _force_into_codepoints(cls, other):
         assert not is_tracing()
+        other = typeable_value(other)
         if isinstance(other, LazyIntSymbolicStr):
             return other._codepoints
         elif isinstance(other, str):

--- a/crosshair/libimpl/builtinslib_test.py
+++ b/crosshair/libimpl/builtinslib_test.py
@@ -69,7 +69,12 @@ from crosshair.statespace import (
     POST_FAIL,
     MessageType,
 )
-from crosshair.test_util import check_exec_err, check_states, summarize_execution
+from crosshair.test_util import (
+    check_exec_err,
+    check_messages,
+    check_states,
+    summarize_execution,
+)
 from crosshair.tracers import NoTracing, ResumedTracing
 from crosshair.util import CrosshairInternal, IgnoreAttempt, set_debug
 
@@ -2797,6 +2802,22 @@ def test_object_with_comparison():
         return obj != b"abc"
 
     check_states(f, POST_FAIL)
+
+
+def test_object_addition():
+    def f(a, b):
+        """post: True"""
+        return a + b
+
+    actual, expected = check_messages(
+        analyze_function(f),
+        state=MessageType.EXEC_ERR,
+        # We check the message here, because we used to produce an
+        # (incorrect) counterexample of f('', '')
+        # See https://github.com/pschanely/CrossHair/issues/235
+        message="TypeError:  when calling f('', 0)",
+    )
+    assert actual == expected
 
 
 def test_callable_zero_args() -> None:

